py/selenium/webdriver/chromium/virtual_authenticator.py

```diff
@@ -0,0 +1,248 @@
+        self._is_user_verified = value
+
+    def to_dict(self) -> dict:
+        return {
+            "protocol": self.protocol,
+            "transport": self.transport,
+            "hasResidentKey": self.has_resident_key,
+            "hasUserVerification": self.has_user_verification,
+            "isUserConsenting": self.is_user_consenting,
+            "isUserVerified": self.is_user_verified
+        }
+
+
+class Credential:
+    def __init__(self, credential_id: bytes, is_resident_credential: bool, rp_id: str, user_handle: bytes, private_key: bytes, sign_count: int):
+        """Constructor. A credential stored in a virtual authenticator.
+        https://w3c.github.io/webauthn/#credential-parameters
+
+        :Args:
+            - credential_id (bytes): Unique base64 encoded string.
+            is_resident_credential (bool): Whether the credential is client-side discoverable.
+            rp_id (str): Relying party identifier.
+            user_handle (bytes): userHandle associated to the credential. Must be Base64 encoded string. Can be None.
+            private_key (bytes): Base64 encoded PKCS#8 private key.
+            sign_count (int): intital value for a signature counter.
+        """
+        self._id= credential_id
+        self._is_resident_credential = is_resident_credential
+        self._rp_id = rp_id
+        self._user_handle = user_handle
+        self._private_key = private_key
+        self._sign_count = sign_count
+
+    @property
+    def id(self):
+        return self._id
+
+    @id.getter
+    def id(self):
+        return urlsafe_b64encode(self._id).decode()
+
+    @property
+    def is_resident_credential(self):
+        return self._is_resident_credential
+
+    @property
+    def rp_id(self):
+        return self._rp_id
+
+    @property
+    def user_handle(self):
+        return self._user_handle
+
+    @user_handle.getter
+    def user_handle(self):
+        if self._user_handle:
+            return urlsafe_b64encode(self._user_handle).decode()
+        return None
+
+    @property
+    def private_key(self):
+        return self._private_key
+
+    @private_key.getter
+    def private_key(self):
+        return urlsafe_b64encode(self._private_key).decode()
+
+    @property
+    def sign_count(self):
+        return self._sign_count
+
+    @staticmethod
+    def create_non_resident_credential(id: bytes,  rp_id: str, private_key: bytes, sign_count: int) -> 'Credential':
+        """Creates a non-resident (i.e. stateless) credential.
+
+              :Args:
+                - id (bytes): Unique base64 encoded string.
+                - rp_id (str): Relying party identifier.
+                - private_key (bytes): Base64 encoded PKCS
+                - sign_count (int): intital value for a signature counter.
+
+              :Returns:
+                - Credential: A non-resident credential.
+        """
+        return Credential(id, False, rp_id, None, private_key, sign_count)
+
+    @staticmethod
```

Comment:
prefer `@classmethod` for these for similar reasons mentioned on the `from_dict(...)`

Reply:
thanks